modules/motions/hooks/useAvailableMotions.ts

```diff
@@ -22,113 +22,128 @@ const isHasTrustedCaller = (
+    walletAddress && nodeOperators && sandboxNodeOperators
+      ? `available-motions-${chainId}-${walletAddress}`
+      : null,
+    async () => {
+      if (
+        !walletAddress ||
+        nodeOperators === undefined ||
+        sandboxNodeOperators === undefined
+      ) {
+        return {} as AvailableMotions
+      }
+
+      const parsedChainId = parseEvmSupportedChainId(chainId)
+      const nodeOperatorIncreaseLimitAddress =
+        EVM_CONTRACTS[MotionTypeForms.NodeOperatorIncreaseLimit].address[
+          parsedChainId
+        ]
+      const sandboxNodeOperatorIncreaseLimitAddress =
+        EVM_CONTRACTS[MotionTypeForms.SandboxNodeOperatorIncreaseLimit].address[
+          parsedChainId
+        ]
+
+      const promiseResult = await Promise.allSettled(
+        Object.values(EVM_CONTRACTS).map(async contract => {
+          const contractAddress = contract.address[chainId]
+
+          if (
+            !contractAddress ||
+            contractAddress === nodeOperatorIncreaseLimitAddress ||
+            contractAddress === sandboxNodeOperatorIncreaseLimitAddress
+          ) {
+            return null
+          }
+
+          const connectedContract = contract.connectRpc({ chainId })
+
+          if (!isHasTrustedCaller(connectedContract)) {
+            return null
+          }
+
+          const trustedCaller = await connectedContract.trustedCaller()
+
+          return { contractAddress, trustedCaller }
+        }),
+      )
+
+      const isNodeOperatorConnected = getIsNodeOperatorConnected(
+        walletAddress,
+        nodeOperators,
+      )
+      const isSandboxNodeOperatorConnected = getIsNodeOperatorConnected(
+        walletAddress,
+        sandboxNodeOperators,
+      )
+
+      return promiseResult.reduce(
+        (acc, cur) => {
+          if (cur.status !== 'fulfilled' || !cur.value) {
+            return acc
+          }
+
+          const { contractAddress, trustedCaller } = cur.value
+
+          const contractType =
+            EvmTypesByAdress[parseEvmSupportedChainId(chainId)][contractAddress]
+
+          if (
+            !contractType ||
+            !Object.keys(MotionTypeForms).includes(contractType)
+          ) {
+            return acc
+          }
+
+          acc[contractType as MotionTypeForms] = trustedCaller === walletAddress
+
+          return acc
+        },
+        {
+          [MotionTypeForms.NodeOperatorIncreaseLimit]: isNodeOperatorConnected,
+          [MotionTypeForms.SandboxNodeOperatorIncreaseLimit]:
+            isSandboxNodeOperatorConnected,
+        } as AvailableMotions,
+      )
+    },
+    { revalidateOnReconnect: false, revalidateOnFocus: false },
+  )
 
   const notHaveAvailableMotions = useMemo(() => {
     if (!availableMotions) return true
     return Object.values(availableMotions).every(value => !value)
   }, [availableMotions])
 
-  return { availableMotions, notHaveAvailableMotions }
+  return {
+    availableMotions,
+    isLoading:
```

Comment:
May be better to name key `initialLoading` because it is combination of 3 different `initialLoading`